File: ClientApp/src/app/models/lobby.ts
import {User} from './user';

export class Lobby {
    lobbyId: number;
    name: string;
    hostId: string;
    users: User[];
    lobbyUser: LobbyUser[];

    constructor(name: string, host: User)
    {
        this.name = name;
        this.hostId = host.userId;
    }
}

export class LobbyUser {
    id: number;
    lobbyId: number;
    userId: string;
}

File: Models/Lobby.cs
using System;
using System.Collections.Generic;

namespace DominionWeb.Models
{
    public partial class Lobby
    {
        public Lobby()
        {
            LobbyUser = new HashSet<LobbyUser>();
        }

        public int LobbyId { get; set; }
        public string Name { get; set; }
        public string HostId { get; set; }

        public virtual ICollection<LobbyUser> LobbyUser { get; set; }
    }
}
